cmd/snap-confine/user-support.c

```diff
@@ -37,6 +38,11 @@ void setup_user_data(void)
 
 	debug("creating user data directory: %s", user_data);
 	if (sc_nonfatal_mkpath(user_data, 0755) < 0) {
+		if (errno == EROFS && !sc_startswith(user_data, "/home")) {
```

Comment:
This would misbehave for /home2/bob. Please add a trailing slash:

```suggestion
		if (errno == EROFS && !sc_startswith(user_data, "/home/")) {
```